provider/parameter.go

```diff
@@ -331,6 +335,11 @@ func parameterDataSource() *schema.Resource {
 				Computed:    true,
 				Description: "Whether this value is optional.",
 			},
+			"priority": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The priority of template parameters determines their position ahead of parameters with lower priorities in the UI/CLI presentation.",
```

Comment:
Maybe we can define the edge cases here a bit more?

```suggestion
				Description: "The priority of a template parameter determines its position in the UI/CLI presentation. A higher priority parameter is positioned ahead of a lower priority one and parameters with the same priority are ordered by name in ascending order.",
```

Reply:
Done!